sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -77,9 +77,10 @@ abstract class JdbcDialect {
   /**
    * Retrieve the jdbc / sql type for a given datatype.
    * @param dt The datatype (e.g. [[org.apache.spark.sql.types.StringType]])
+   * @param md The metadata
    * @return The new JdbcType if there is an override for this DataType
    */
-  def getJDBCType(dt: DataType): Option[JdbcType] = None
+  def getJDBCType(dt: DataType, md: Metadata): Option[JdbcType] = None
```

Comment:
While this is a `DeveloperAPI`, it is public so it would be good to fix without breaking binary compatibility.